lab_5_scrapper/scrapper.py

```diff
@@ -120,45 +263,141 @@ def __init__(self, full_url: str, article_id: int, config: Config) -> None:
         """
         Initializes an instance of the HTMLParser class
         """
-        pass
+        self.full_url = full_url
+        self.article_id = article_id
+        self.config = config
+        self.article = Article(self.full_url, self.article_id)
 
     def _fill_article_with_text(self, article_soup: BeautifulSoup) -> None:
         """
         Finds text of article
         """
-        pass
+        text_elements = article_soup.find("div", class_="page-main__text").find_all("p")
+        self.article.text = "\n".join([p.get_text(strip=True) for p in text_elements])
 
     def _fill_article_with_meta_information(self, article_soup: BeautifulSoup) -> None:
         """
         Finds meta information of article
         """
-        pass
+        author_elem = article_soup.find_all('a', class_='page-main__publish-author global-link')[0]
+        authors = [author_elem.get_text(strip=True)] if author_elem else ["NOT FOUND"]
+
+        date_elem = article_soup.find_all('a', class_='page-main__publish-date')[0]
+        date_str = date_elem.get_text(strip=True) if date_elem else "NOT FOUND"
+        date = self.unify_date_format(date_str)
+
+        topics_elem = article_soup.find_all('a', class_='panel-group__title global-link')[1]
+        topic = topics_elem.get_text(strip=True) if topics_elem else "NOT FOUND"
+
+        title_elem = article_soup.find_all('h1', class_='page-main__head')[0]
+        title = title_elem.get_text(strip=True) if title_elem else "NOT FOUND"
+
+        self.article.author = authors
+        self.article.date = date
+        self.article.topics = topic
+        self.article.title = title
 
     def unify_date_format(self, date_str: str) -> datetime.datetime:
         """
         Unifies date format
         """
-        pass
+        months_dict = {
+            "января": "January",
+            "февраля": "February",
+            "марта": "March",
+            "апреля": "April",
+            "мая": "May",
+            "июня": "June",
+            "июля": "July",
+            "августа": "August",
+            "сентября": "September",
+            "октября": "October",
+            "ноября": "November",
+            "декабря": "December"
+        }
+        date_str = date_str.replace(date_str.split()[1], months_dict[date_str.split()[1]])
+        date_obj = str(datetime.datetime.strptime(date_str, '%d %B %Y - %H:%M'))
```

Comment:
why do you make parse twice? you should stop as soon as you get datetime instance and return it to be assigned to article attribute